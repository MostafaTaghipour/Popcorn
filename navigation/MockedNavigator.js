import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import { Provider as ReduxProvider } from "react-redux";
import { AuthProvider } from "../contexts/AuthContext";

const Stack = createStackNavigator();

/**
 * Provide a MockedNavigator to wrap our screens
 * 
 * Access navigation in screens 
 * 
 * @param {*} param0 
 */
export const MockedNavigator = ({ component, params = {}, store }) => {
  return (
    <ReduxProvider store={store}>
      <AuthProvider>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen
              name="MockedScreen"
              component={component}
              initialParams={params}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </AuthProvider>
    </ReduxProvider>
  );
};
