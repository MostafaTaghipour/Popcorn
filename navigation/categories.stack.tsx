import React from "react";
import {
  createStackNavigator,
  StackNavigationProp,
} from "@react-navigation/stack";
import { RouteProp } from "@react-navigation/native";
import { SharedStackParamList } from "./SharedNavParamList";
import CategoriesScreen from "@app/screens/Categories/CategoriesScreen";
import CategoryScreen from "@app/screens/Category/CategoryScreen";

interface CategoriesStackProps {}

type CategoriesStackParamList = {
  Categories: undefined;
  Category: {
    title: string;
  };
} & SharedStackParamList;

export type CategoriesStackNavProps<
  T extends keyof CategoriesStackParamList
> = {
  navigation: StackNavigationProp<CategoriesStackParamList, T>;
  route: RouteProp<CategoriesStackParamList, T>;
};

const Stack = createStackNavigator<CategoriesStackParamList>();


/**
 * Categories Stack contains CategoriesScreen and CategoryScreen
 *
 * @return {*} 
 */
export const CategoriesStack: React.FC<CategoriesStackProps> = () => {
  return (
    <Stack.Navigator >
      <Stack.Screen
        name="Categories"
        component={CategoriesScreen}
      ></Stack.Screen>
        <Stack.Screen
        name="Category"
        component={CategoryScreen}
      ></Stack.Screen>
    </Stack.Navigator>
  );
};
