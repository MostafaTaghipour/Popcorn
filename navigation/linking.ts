import { LinkingOptions } from "@react-navigation/native/lib/typescript/src/types";
/**
 * If you are using universal links, you need to add your domain to the prefixes.
 * https://reactnavigation.org/docs/deep-linking/
 * 
 *  @type {*} */
const options: LinkingOptions = {
  prefixes: ["https://popcorn.com", "popcorn://"],
  // config: {
  // },
};
export default options;
