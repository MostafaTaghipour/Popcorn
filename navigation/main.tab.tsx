import React, { useContext } from "react";
import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
import { Feather } from "@expo/vector-icons";
import { HomeStack } from "./home.stack";
import { LocalizationContext } from "@app/contexts/LocalizationContext";
import { CategoriesStack } from "./categories.stack";
import { SettingsStack } from "./settings.stack";

interface MainTabProps {}

type MainTabParamList = {
  Home: undefined;
  Categories: undefined;
  Settings: undefined;
};

const Tab = createBottomTabNavigator<MainTabParamList>();



/**
 * MainTabs contains HomeStack , CategoriesStack and Settings
 *
 * @param {*} {}
 * @return {*} 
 */
export const MainTabs: React.FC<MainTabProps> = ({}) => {
  const { t } = useContext(LocalizationContext);

  return (
    <Tab.Navigator tabBarOptions={{ showLabel: false }}>
      <Tab.Screen
        name="Home"
        component={HomeStack}
        options={{
          title: t("home.title"),
          tabBarIcon: ({ color }) => {
            return <Feather name="home" size={24} color={color} />;
          },
        }}
      ></Tab.Screen>
      <Tab.Screen
        name="Categories"
        component={CategoriesStack}
        options={{
          title: t("categories.title"),
          tabBarIcon: ({ color }) => {
            return <Feather name="grid" size={24} color={color} />;
          },
        }}
      ></Tab.Screen>
      <Tab.Screen
        name="Settings"
        component={SettingsStack}
        options={{
          title: t("settings.title"),
          tabBarIcon: ({ color }) => {
            return <Feather name="settings" size={24} color={color} />;
          },
        }}
      ></Tab.Screen>
    </Tab.Navigator>
  );
};
