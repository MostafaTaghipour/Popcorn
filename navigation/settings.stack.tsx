import React from "react";
import {
  createStackNavigator,
  StackNavigationProp,
} from "@react-navigation/stack";
import { RouteProp } from "@react-navigation/native";
import { SharedStackParamList } from "./SharedNavParamList";
import SettingsScreen from "@app/screens/Settings/SettingsScreen";

interface SettingsStackProps {}

type SettingsStackParamList = {
  Settings: undefined;
} & SharedStackParamList;

export type SettingsStackNavProps<T extends keyof SettingsStackParamList> = {
  navigation: StackNavigationProp<SettingsStackParamList, T>;
  route: RouteProp<SettingsStackParamList, T>;
};

const Stack = createStackNavigator<SettingsStackParamList>();

/**
 * SettingsStack contains SettingsScreen 
 *
 * @return {*} 
 */
export const SettingsStack: React.FC<SettingsStackProps> = ({
  navigation,
  route,
}: any) => {
  navigation.setOptions({
    tabBarVisible: route.state ? (route.state.index > 0 ? false : true) : null,
  });

  return (
    <Stack.Navigator>
      <Stack.Screen name="Settings" component={SettingsScreen}></Stack.Screen>
    </Stack.Navigator>
  );
};
