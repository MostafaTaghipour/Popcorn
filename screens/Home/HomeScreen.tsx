import { StyleSheet, View, SafeAreaView, Image } from "react-native";
import React, { useContext, useEffect, useState } from "react";
import {
  Title,
  Text,
  Caption,
  TouchableRipple,
  Card,
} from "react-native-paper";
import { HomeStackNavProps } from "@app/navigation/home.stack";
import { LocalizationContext } from "@app/contexts/LocalizationContext";
import { AuthContext } from "@app/contexts/AuthContext";
import { Feather } from "@expo/vector-icons";
import { ScrollView } from "react-native-gesture-handler";
import Carousel from "@app/components/Carousel";
import { mockMovies } from "@app/data/mock/movie";
import { Movie } from "@app/store/movie/types";
import SlideShow from "@app/components/SlideShow";
import Layout from "@app/constants/Layout";
import { useScrollToTop } from "@react-navigation/native";
import { ThemeContext } from "@app/contexts/ThemeContext";
import Colors from "@app/constants/Colors";

/**
 * Our Home page
 * to have a nice home page we use some static fake data
 *
 * @export
 * @param {HomeStackNavProps<"Home">} {
 *   navigation,
 * }
 * @return {*}
 */
export default function HomeScreen({ navigation }: HomeStackNavProps<"Home">) {
  const { t } = useContext(LocalizationContext);
  const { isDark, theme } = useContext(ThemeContext);
  const { isAuthenticated, userName } = useContext(AuthContext);
  const scrollRef = React.useRef(null);
  const [bestMovies, setBestMovies] = useState<Movie[]>([]);
  const [popularMovies, setPopularMovies] = useState<Movie[]>([]);
  const [topMovies, setTopMovies] = useState<Movie[]>([]);

  /**
   * scroll to top when tapping on the active tab
   * https://reactnavigation.org/docs/use-scroll-to-top/
   *
   */
  useScrollToTop(scrollRef);

  /**
   * to have a nice home page we use some static fake data
   */
  useEffect(() => {
    setBestMovies(mockMovies.slice(0, 3));
    setPopularMovies(mockMovies.slice(3, 7));
    setTopMovies(mockMovies.slice(7, 11));
  }, []);

  /**
   * hide header
   */
  navigation.setOptions({
    headerShown: false,
  });

  /**
   * navigate to search screen
   *
   */
  const search = () => {
    navigation.navigate("Search");
  };

  /**
   * navigate to auth screen
   *
   */
  const login = () => {
    navigation.navigate("Auth");
  };

  return (
    <ScrollView ref={scrollRef}>
      <SafeAreaView style={styles.container}>
        <View style={styles.topSection}>
          <Title>
            {t("home.welcome", { name: userName || t("home.user") })}
          </Title>
        </View>
        {!isAuthenticated && (
          <TouchableRipple onPress={login} testID="UserLoginSection">
            <View style={styles.loginSection}>
              <Caption>{t("home.please")}</Caption>
              <Caption
                style={{
                  color: theme.colors.primary,
                  fontFamily: theme.fonts.medium.fontFamily,
                  marginHorizontal: 4,
                }}
              >
                {t("home.login")}
              </Caption>
              <Caption>{t("home.to_your_account")}</Caption>
            </View>
          </TouchableRipple>
        )}

        <TouchableRipple
          style={[
            styles.searchBoxButton,
            { backgroundColor: isDark ? Colors.nero : Colors.gainsboro },
          ]}
          onPress={search}
        >
          <View style={styles.searchBox}>
            <Text style={styles.searchBoxLabel}>{t("home.search_hint")}</Text>
            <Feather name="search" size={20} color="gray" />
          </View>
        </TouchableRipple>

        <SlideShow
          data={bestMovies}
          keyExtractor={(item) => item.id.toString()}
          renderItem={({ item }) => {
            return <SlideShowItem movie={item} />;
          }}
          style={styles.slideShow}
        />

        <Carousel
          title={t("home.popular")}
          data={popularMovies}
          contentContainerStyle={styles.carousel}
          keyExtractor={(item) => item.id.toString()}
          renderItem={({ item }) => {
            return <CarouselItem movie={item} />;
          }}
        />

        <Carousel
          title={t("home.top")}
          data={topMovies}
          contentContainerStyle={styles.carousel}
          keyExtractor={(item) => item.id.toString()}
          renderItem={({ item }) => {
            return <CarouselItem movie={item} />;
          }}
        />
      </SafeAreaView>
    </ScrollView>
  );
}

/**
 * CarouselItem , pure component
 *
 *  @type {*} */
const CarouselItem = React.memo(({ movie }: { movie: Movie }) => {
  return (
    <Card style={styles.carouselItem}>
      <Image
        style={styles.carouselItemImage}
        source={{ uri: movie.poster }}
      ></Image>
    </Card>
  );
});

/**
 * SlideShowItem , pure component
 *
 *  @type {*} */
const SlideShowItem = React.memo(({ movie }: { movie: Movie }) => {
  return (
    <View style={styles.slideShowItem}>
      <Card style={styles.slideShowCard}>
        <Image
          style={styles.carouselItemImage}
          source={{ uri: movie.poster }}
        ></Image>
      </Card>
    </View>
  );
});

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  topSection: {
    flexDirection: "row",
    justifyContent: "space-between",
    marginHorizontal: 16,
    marginTop: 16,
  },
  loginSection: {
    marginTop: -8,
    flexDirection: "row",
    marginHorizontal: 16,
    paddingVertical: 8,
  },
  searchBoxButton: {
    borderRadius: 23,
    margin: 16,
    height: 45,
  },
  searchBox: {
    justifyContent: "space-between",
    alignItems: "center",
    flexDirection: "row",
    flex: 1,
    paddingHorizontal: 16,
  },
  searchBoxLabel: {
    color: "gray",
  },
  carousel: {
    paddingHorizontal: 8,
    marginBottom: 32,
  },
  carouselItem: {
    width: 150,
    height: 200,
    marginHorizontal: 8,
    overflow: "hidden",
  },
  carouselItemImage: { flex: 1 },

  slideShow: {
    marginVertical: 8,
  },
  slideShowItem: {
    width: Layout.window.width,
    height: 250,
    padding: 16,
  },
  slideShowCard: {
    flex: 1,
    overflow: "hidden",
  },
  slideShowItemImage: { flex: 1 },
});
