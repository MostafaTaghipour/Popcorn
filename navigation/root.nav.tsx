import React, { useContext } from "react";
import { NavigationContainer, RouteProp } from "@react-navigation/native";
import { MainTabs } from "./main.tab";
import linking from "./linking";
import { ThemeContext } from "@app/contexts/ThemeContext";
import { Provider as PaperProvider } from "react-native-paper";
import {
  createStackNavigator,
  StackNavigationProp,
  TransitionPresets,
} from "@react-navigation/stack";
import { SharedStackParamList } from "./SharedNavParamList";
import IntroScreen from "@app/screens/Intro/IntroScreen";
import AuthScreen from "@app/screens/Auth/AuthScreen";
import { StatusBar, Platform } from "react-native";
import { Feather } from "@expo/vector-icons";
import { useReduxSelector } from "@app/store";

type AppNavigatorParamList = {
  MainTab: undefined;
  Intro: undefined;
} & SharedStackParamList;

export type AppNavigatorProps<T extends keyof AppNavigatorParamList> = {
  navigation: StackNavigationProp<AppNavigatorParamList, T>;
  route: RouteProp<AppNavigatorParamList, T>;
};

const Stack = createStackNavigator<AppNavigatorParamList>();

/**
 * RootNavigator handle screen according to seenIntro flag
 *
 * @param {*} {}
 * @return {*}
 */
export function RootNavigator() {
  const { theme, isDark } = useContext(ThemeContext);
  const seenIntro = useReduxSelector((state) => state.ui.seenIntro);

  return (
    <NavigationContainer theme={theme} linking={linking}>
      <PaperProvider theme={theme}>
        {/* Status bar according to current theme */}
        <StatusBar
          barStyle={isDark ? "light-content" : "dark-content"}
          backgroundColor={isDark ? "black" : "white"}
        />
        <Stack.Navigator screenOptions={{ cardOverlayEnabled: !isDark }}>
          {seenIntro ? (
            <>
              <Stack.Screen
                name="MainTab"
                component={MainTabs}
                options={{ headerShown: false }}
              />
              <Stack.Screen
                name="Auth"
                component={AuthScreen}
                options={{
                  headerBackTitleVisible: false,
                  headerBackImage: () => (
                    <Feather name="x" size={32} color={theme.colors.primary} />
                  ),
                  gestureEnabled: true,
                  cardOverlayEnabled: true,
                  ...Platform.select({
                    ios: { ...TransitionPresets.ModalPresentationIOS },
                    android: { ...TransitionPresets.ModalTransition },
                  }),
                  headerShown: false,
                }}
              />
            </>
          ) : (
            <Stack.Screen
              name="Intro"
              component={IntroScreen}
              options={{ headerShown: false }}
            />
          )}
        </Stack.Navigator>
      </PaperProvider>
    </NavigationContainer>
  );
}
