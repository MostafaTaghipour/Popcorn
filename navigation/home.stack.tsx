import React from "react";
import {
  createStackNavigator,
  StackNavigationProp,
} from "@react-navigation/stack";
import { RouteProp } from "@react-navigation/native";
import { SharedStackParamList } from "./SharedNavParamList";
import HomeScreen from "@app/screens/Home/HomeScreen";
import SearchScreen from "@app/screens/Search/SearchScreen";

interface HomeStackProps {}

type HomeStackParamList = {
  Home: undefined;
  Search: undefined;
} & SharedStackParamList;

export type HomeStackNavProps<T extends keyof HomeStackParamList> = {
  navigation: StackNavigationProp<HomeStackParamList, T>;
  route: RouteProp<HomeStackParamList, T>;
};

const Stack = createStackNavigator<HomeStackParamList>();

/**
 * HomeStack contains HomeScreen and SearchScreen
 *
 * @return {*} 
 */
export const HomeStack: React.FC<HomeStackProps> = ({
  navigation,
  route,
}: any) => {
  navigation.setOptions({
    tabBarVisible: route.state ? (route.state.index > 0 ? false : true) : null,
  });
  return (
    <Stack.Navigator >
      <Stack.Screen name="Home" component={HomeScreen}></Stack.Screen>
      <Stack.Screen name="Search" component={SearchScreen}></Stack.Screen>
    </Stack.Navigator>
  );
};
