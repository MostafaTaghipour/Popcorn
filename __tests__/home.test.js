import React from "react";
import HomeScreen from "../screens/Home/HomeScreen";
import { MockedNavigator } from "../navigation/MockedNavigator";
import { render } from "react-native-testing-library";
import configureStore from "redux-mock-store";

/**
 * test HomeScreen rendering
 */
describe("Home render", () => {
  let initialState;
  let store;

  /**
   * initialize mock redux store
   */
  beforeAll(() => {
    const middlewares = [];
    const mockStore = configureStore(middlewares);

    // Initialize mockstore
    initialState = {
      auth: {
        token: undefined,
      },
    };
    store = mockStore(initialState);
  });

  /**
   * Test HomeScreen snapshot
   */
  test("Home snapshot", () => {
    const snap = render(
      <MockedNavigator component={HomeScreen} store={store} />
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });

  /**
   * If user is logged in UserLoginSection must  be shown
   */
  it("Testing UserLoginSection visibility", () => {
    const tree = render(
      <MockedNavigator component={HomeScreen} store={store} />
    );

    expect(tree.getAllByTestId("UserLoginSection").length).toBe(1);
  });
});
