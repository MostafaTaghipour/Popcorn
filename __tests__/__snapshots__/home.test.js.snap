// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render Home snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTScrollView
                      collapsable={false}
                      onGestureHandlerEvent={[Function]}
                      onGestureHandlerStateChange={[Function]}
                    >
                      <View>
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                                "marginHorizontal": 16,
                                "marginTop": 16,
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#000000",
                                    "fontFamily": "System",
                                    "fontWeight": "400",
                                    "textAlign": "left",
                                  },
                                  Array [
                                    Object {
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "fontFamily": "System",
                                      "fontWeight": "500",
                                      "textAlign": "left",
                                      "writingDirection": "ltr",
                                    },
                                    Array [
                                      Object {
                                        "fontSize": 20,
                                        "letterSpacing": 0.15,
                                        "lineHeight": 30,
                                        "marginVertical": 2,
                                      },
                                      undefined,
                                    ],
                                    Array [
                                      Object {
                                        "fontSize": 20,
                                        "letterSpacing": 0.15,
                                        "lineHeight": 30,
                                        "marginVertical": 2,
                                      },
                                      undefined,
                                    ],
                                  ],
                                ]
                              }
                            >
                              
                            </Text>
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            isTVSelectable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                false,
                                undefined,
                              ]
                            }
                            testID="UserLoginSection"
                          >
                            <View
                              style={
                                Object {
                                  "flexDirection": "row",
                                  "marginHorizontal": 16,
                                  "marginTop": -8,
                                  "paddingVertical": 8,
                                }
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#000000",
                                      "fontFamily": "System",
                                      "fontWeight": "400",
                                      "textAlign": "left",
                                    },
                                    Array [
                                      Object {
                                        "color": "rgba(0, 0, 0, 0.54)",
                                        "fontFamily": "System",
                                        "fontWeight": "400",
                                        "textAlign": "left",
                                        "writingDirection": "ltr",
                                      },
                                      Array [
                                        Object {
                                          "fontSize": 12,
                                          "letterSpacing": 0.4,
                                          "lineHeight": 20,
                                          "marginVertical": 2,
                                        },
                                        undefined,
                                      ],
                                      Array [
                                        Object {
                                          "fontSize": 12,
                                          "letterSpacing": 0.4,
                                          "lineHeight": 20,
                                          "marginVertical": 2,
                                        },
                                        undefined,
                                      ],
                                    ],
                                  ]
                                }
                              >
                                
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#000000",
                                      "fontFamily": "System",
                                      "fontWeight": "400",
                                      "textAlign": "left",
                                    },
                                    Array [
                                      Object {
                                        "color": "rgba(0, 0, 0, 0.54)",
                                        "fontFamily": "System",
                                        "fontWeight": "400",
                                        "textAlign": "left",
                                        "writingDirection": "ltr",
                                      },
                                      Array [
                                        Object {
                                          "fontSize": 12,
                                          "letterSpacing": 0.4,
                                          "lineHeight": 20,
                                          "marginVertical": 2,
                                        },
                                        Object {
                                          "color": "#ff9900",
                                          "fontFamily": "System",
                                          "marginHorizontal": 4,
                                        },
                                      ],
                                      Array [
                                        Object {
                                          "fontSize": 12,
                                          "letterSpacing": 0.4,
                                          "lineHeight": 20,
                                          "marginVertical": 2,
                                        },
                                        Object {
                                          "color": "#ff9900",
                                          "fontFamily": "System",
                                          "marginHorizontal": 4,
                                        },
                                      ],
                                    ],
                                  ]
                                }
                              >
                                
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#000000",
                                      "fontFamily": "System",
                                      "fontWeight": "400",
                                      "textAlign": "left",
                                    },
                                    Array [
                                      Object {
                                        "color": "rgba(0, 0, 0, 0.54)",
                                        "fontFamily": "System",
                                        "fontWeight": "400",
                                        "textAlign": "left",
                                        "writingDirection": "ltr",
                                      },
                                      Array [
                                        Object {
                                          "fontSize": 12,
                                          "letterSpacing": 0.4,
                                          "lineHeight": 20,
                                          "marginVertical": 2,
                                        },
                                        undefined,
                                      ],
                                      Array [
                                        Object {
                                          "fontSize": 12,
                                          "letterSpacing": 0.4,
                                          "lineHeight": 20,
                                          "marginVertical": 2,
                                        },
                                        undefined,
                                      ],
                                    ],
                                  ]
                                }
                              >
                                
                              </Text>
                            </View>
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            isTVSelectable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                false,
                                Array [
                                  Object {
                                    "borderRadius": 23,
                                    "height": 45,
                                    "margin": 16,
                                  },
                                  Object {
                                    "backgroundColor": "#e2e2e2",
                                  },
                                ],
                              ]
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flex": 1,
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                  "paddingHorizontal": 16,
                                }
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#000000",
                                      "fontFamily": "System",
                                      "fontWeight": "400",
                                      "textAlign": "left",
                                    },
                                    Object {
                                      "color": "gray",
                                    },
                                  ]
                                }
                              >
                                
                              </Text>
                              <Text />
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "marginVertical": 8,
                              }
                            }
                          >
                            <RCTScrollView
                              disableVirtualization={false}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={true}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              numColumns={1}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              pagingEnabled={true}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEventThrottle={50}
                              showsHorizontalScrollIndicator={false}
                              stickyHeaderIndices={Array []}
                              style={
                                Object {
                                  "marginVertical": 8,
                                }
                              }
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View />
                            </RCTScrollView>
                          </View>
                          <View>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#000000",
                                    "fontFamily": "System",
                                    "fontWeight": "400",
                                    "textAlign": "left",
                                  },
                                  Array [
                                    Object {
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "fontFamily": "System",
                                      "fontWeight": "500",
                                      "textAlign": "left",
                                      "writingDirection": "ltr",
                                    },
                                    Array [
                                      Object {
                                        "fontSize": 20,
                                        "letterSpacing": 0.15,
                                        "lineHeight": 30,
                                        "marginVertical": 2,
                                      },
                                      Object {
                                        "color": "gray",
                                        "fontSize": 16,
                                        "marginStart": 16,
                                      },
                                    ],
                                    Array [
                                      Object {
                                        "fontSize": 20,
                                        "letterSpacing": 0.15,
                                        "lineHeight": 30,
                                        "marginVertical": 2,
                                      },
                                      Object {
                                        "color": "gray",
                                        "fontSize": 16,
                                        "marginStart": 16,
                                      },
                                    ],
                                  ],
                                ]
                              }
                            >
                              
                            </Text>
                            <RCTScrollView
                              contentContainerStyle={
                                Object {
                                  "marginBottom": 32,
                                  "paddingHorizontal": 8,
                                }
                              }
                              disableVirtualization={false}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={true}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              numColumns={1}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEventThrottle={50}
                              showsHorizontalScrollIndicator={false}
                              stickyHeaderIndices={Array []}
                              title=""
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View />
                            </RCTScrollView>
                          </View>
                          <View>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#000000",
                                    "fontFamily": "System",
                                    "fontWeight": "400",
                                    "textAlign": "left",
                                  },
                                  Array [
                                    Object {
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "fontFamily": "System",
                                      "fontWeight": "500",
                                      "textAlign": "left",
                                      "writingDirection": "ltr",
                                    },
                                    Array [
                                      Object {
                                        "fontSize": 20,
                                        "letterSpacing": 0.15,
                                        "lineHeight": 30,
                                        "marginVertical": 2,
                                      },
                                      Object {
                                        "color": "gray",
                                        "fontSize": 16,
                                        "marginStart": 16,
                                      },
                                    ],
                                    Array [
                                      Object {
                                        "fontSize": 20,
                                        "letterSpacing": 0.15,
                                        "lineHeight": 30,
                                        "marginVertical": 2,
                                      },
                                      Object {
                                        "color": "gray",
                                        "fontSize": 16,
                                        "marginStart": 16,
                                      },
                                    ],
                                  ],
                                ]
                              }
                            >
                              
                            </Text>
                            <RCTScrollView
                              contentContainerStyle={
                                Object {
                                  "marginBottom": 32,
                                  "paddingHorizontal": 8,
                                }
                              }
                              disableVirtualization={false}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={true}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              numColumns={1}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEventThrottle={50}
                              showsHorizontalScrollIndicator={false}
                              stickyHeaderIndices={Array []}
                              title=""
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View />
                            </RCTScrollView>
                          </View>
                        </RCTSafeAreaView>
                      </View>
                    </RCTScrollView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      />
    </View>
  </RNCSafeAreaView>
</View>
`;
